### PR TITLE
DAOS-7829 doc: Added Common Tasks section in docs/admin

### PR DESCRIPTION
**Pull request by shimizukko**
Added Common Tasks that describes the common tasks handled by admins.

Doc-only: true
Signed-off-by: Makito Kano <makito.kano@intel.com>

### CODE REVIEW

File: docs/admin/common_tasks.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
+# DAOS Common Tasks
+
+This section describes some of the common tasks handled by admins at a high level. See [System Deployment](./deployment#system-deployment), [DAOS System Administration](./administration.md#daos-system-administration), and [Pool Operations](./pool_operations#pool-operations) for more detailed explanations about each step.
+
+## Single host setup with PMEM and NVMe
+
+1. Check PMEM and NVMe are discovered by the system. Format and reset them.
+2. Check network configuration. Check that `ib` interfaces are active.
+3. Install `daos-server` and `daos-client` RPM.
```
Comment (daltonbohning) [inline docs/admin/common_tasks.md:9]:
```suggestion
3. Install `daos-server` and `daos-client` RPMs.
```

File: docs/admin/common_tasks.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+4. Generate certificate files.
+5. Place one of the example configs from `utils/config/examples` to
+`/etc/daos` and adjust it based on the environment. e.g., `access_points`,
```
Comment (daltonbohning) [inline docs/admin/common_tasks.md:12]:
```suggestion
`/etc/daos` and adjust it based on the environment. E.g., `access_points`,
```

File: docs/admin/common_tasks.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+This section describes some of the common tasks handled by admins at a high level. See [System Deployment](./deployment#system-deployment), [DAOS System Administration](./administration.md#daos-system-administration), and [Pool Operations](./pool_operations#pool-operations) for more detailed explanations about each step.
+
+## Single host setup with PMEM and NVMe
+
+1. Check PMEM and NVMe are discovered by the system. Format and reset them.
+2. Check network configuration. Check that `ib` interfaces are active.
+3. Install `daos-server` and `daos-client` RPM.
+4. Generate certificate files.
+5. Place one of the example configs from `utils/config/examples` to
```
Comment (daltonbohning) [inline docs/admin/common_tasks.md:11]: Maybe this is better wording
```suggestion
5. Copy one of the example configs from `utils/config/examples` to
```

File: docs/admin/common_tasks.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+16. Add a large file that's less than the 500MB pool size into the container.
+17. Call `dmg pool query` and check that the free size has declined.
+
+## Multiple host setup with PMEM and NVMe
+
+1. Check PMEM and NVMe configurations are homogeneous. i.e., same number of
+disks, size, address, etc.
+2. Check network configuration. Check that both the server and the client hosts
+can communicate with the Infiniband interface.
+3. Install the same version of `daos-server` and `daos-client` RPM to all the
```
Comment (daltonbohning) [inline docs/admin/common_tasks.md:34]:
```suggestion
3. Install the same version of `daos-server` and `daos-client` RPMs to all the
```

File: docs/admin/common_tasks.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+hosts.
+4. Generate certificate files and distribute them to all the hosts.
+5. Place one of the example configs from `utils/config/examples` to
+`/etc/daos` of one of the server hosts and adjust it based on the environment.
+e.g., `access_points`, `bdev_class`.
```
Comment (daltonbohning) [inline docs/admin/common_tasks.md:39]:
```suggestion
E.g., `access_points`, `bdev_class`.
```

File: docs/admin/common_tasks.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+6. Start `daos_server`.
+7. Use dmg config generate to generate the config file that contains PMEM and
+NVMe.
+8. Distribute the config file to `/etc/daos` of all hosts.
+9. Start server on all the hosts.
+10. Check that it's waiting for SCM format. Call dmg storage format against all
```
Comment (daltonbohning) [inline docs/admin/common_tasks.md:45]:
```suggestion
10. Check that it's waiting for SCM format. Call `dmg storage format` against all
```

File: docs/admin/common_tasks.md [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+17. Add a large file that's less than the 500MB pool size into the container.
+18. Call dmg pool query and check that the free size has declined.
+
+## Pool size management
+
+1. Start DAOS server with PMEM + NVMe and format.
+2. Call `dmg storage scan --verbose` to get the available size allocated to this
+server. Find the NVMe size.
+3. Create a pool with the size half of the NVMe with `--size`. For example, if
+the total NVMe size allocated is 1.6TB, call
```
Comment (daltonbohning) [inline docs/admin/common_tasks.md:63]: Very recent is now `--size 50%`: https://github.com/daos-stack/daos/commit/81671fd6bf223bb8064ff54e6a757f38b232183d

File: docs/admin/common_tasks.md [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+transport_config:
+  allow_insecure: true
+  ca_cert: /etc/daos/certs/daosCA.crt
+  cert: /etc/daos/certs/admin.crt
+  key: /etc/daos/certs/admin.key
+```
+`server_host` is the hostname where the server is running. `group_name` is
+usually `daos_server`. Also adjust port and `transport_config` accordingly.
+
+3. dmg should be able to talk to the server.
```
Comment (daltonbohning) [inline docs/admin/common_tasks.md:97]:
```suggestion
3. `dmg` should be able to talk to the server.
```

**Review by daltonbohning — COMMENTED** [review]
Mostly just grammar nitpicks from me.

File: docs/admin/common_tasks.md (hunk 1 above)
Comment (shimizukko) [reply docs/admin/common_tasks.md]: Fixed

File: docs/admin/common_tasks.md (hunk 2 above)
Comment (shimizukko) [reply docs/admin/common_tasks.md]: Fixed

File: docs/admin/common_tasks.md (hunk 3 above)
Comment (shimizukko) [reply docs/admin/common_tasks.md]: Fixed

File: docs/admin/common_tasks.md [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+12. Define the certificate files in the agent config.
+13. Start agent.
+14. Create a POSIX container with daos command.
+15. Mount the container with dfuse.
+16. Add a large file that's less than the 500MB pool size into the container.
+17. Call `dmg pool query` and check that the free size has declined.
+
+## Multiple host setup with PMEM and NVMe
+
+1. Check PMEM and NVMe configurations are homogeneous. i.e., same number of
```
[the remark this answers is not in the document]
Comment (shimizukko) [reply docs/admin/common_tasks.md]: Fixed

File: docs/admin/common_tasks.md (hunk 4 above)
Comment (shimizukko) [reply docs/admin/common_tasks.md]: Fixed

File: docs/admin/common_tasks.md [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+disks, size, address, etc.
+2. Check network configuration. Check that both the server and the client hosts
+can communicate with the Infiniband interface.
+3. Install the same version of `daos-server` and `daos-client` RPM to all the
+hosts.
+4. Generate certificate files and distribute them to all the hosts.
+5. Place one of the example configs from `utils/config/examples` to
```
[the remark this answers is not in the document]
Comment (shimizukko) [reply docs/admin/common_tasks.md]: Fixed

File: docs/admin/common_tasks.md (hunk 5 above)
Comment (shimizukko) [reply docs/admin/common_tasks.md]: Fixed

File: docs/admin/common_tasks.md (hunk 6 above)
Comment (shimizukko) [reply docs/admin/common_tasks.md]: Fixed

File: docs/admin/common_tasks.md (hunk 7 above)
Comment (shimizukko) [reply docs/admin/common_tasks.md]: Updated the section to introduce the use of percentage.

File: docs/admin/common_tasks.md [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+1. Start DAOS server on one host.
+2. Create a file that specifies the server host in `/etc/daos`. It's usually
+called `daos_control.yml`. Add the following:
+```
+hostlist:
+- <server_host>
+name: <group_name>
+port: 10001
+transport_config:
+  allow_insecure: true
```
[the remark this answers is not in the document]
Comment (shimizukko) [reply docs/admin/common_tasks.md]: Yes. Updated to false.

File: docs/admin/common_tasks.md [hunk 12]
```diff
@@ -0,0 +1,117 @@
...
+  ca_cert: /etc/daos/certs/daosCA.crt
+  cert: /etc/daos/certs/admin.crt
+  key: /etc/daos/certs/admin.key
+```
+`server_host` is the hostname where the server is running. `group_name` is
+usually `daos_server`. Also adjust port and `transport_config` accordingly.
```
[the remark this answers is not in the document]
Comment (shimizukko) [reply docs/admin/common_tasks.md]: Fixed

File: docs/admin/common_tasks.md (hunk 8 above)
Comment (shimizukko) [reply docs/admin/common_tasks.md]: Fixed

File: docs/admin/common_tasks.md [hunk 13]
```diff
@@ -0,0 +1,105 @@
+# DAOS Common Tasks
+
+This section describes some of the common tasks handled by admins at a high level. See [System Deployment](./deployment#system-deployment), [DAOS System Administration](./administration.md#daos-system-administration), and [Pool Operations](./pool_operations#pool-operations) for more detailed explanations about each step.
```
Comment (daltonbohning) [inline docs/admin/common_tasks.md:3]: Is `.md` needed? When clicking through GitHub, links give 404
```suggestion
This section describes some of the common tasks handled by admins at a high level. See [System Deployment](./deployment.md#system-deployment), [DAOS System Administration](./administration.md#daos-system-administration), and [Pool Operations](./pool_operations.md#pool-operations) for more detailed explanations about each step.
```

Comment (shimizukko) [reply docs/admin/common_tasks.md]: Yes. Thanks for checking.

**Review by daltonbohning — APPROVED** [review]

File: docs/admin/common_tasks.md [hunk 14]
```diff
@@ -0,0 +1,110 @@
...
+7. Start `daos_server`.
+8. Use dmg config generate to generate the config file that contains PMEM and
+NVMe.
+9. Distribute the config file to `/etc/daos` of all hosts.
+10. Start server on all the hosts.
+11. Check that it's waiting for SCM format. Call `dmg storage format` against all
+server hosts.
+12. Check that the servers are running on all the hosts with `dmg system query
+--verbose`.
+13. Create a small pool; ~500MB.
```
Comment (daltonbohning) [inline docs/admin/common_tasks.md:53]:
```suggestion
13. Create a small pool; E.g. ~500MB.
```

File: docs/admin/common_tasks.md [hunk 15]
```diff
@@ -0,0 +1,110 @@
...
+14. Define the certificate files in the agent config in the client host.
+15. Start agent.
+16. Create a POSIX container with daos command.
```
Comment (daltonbohning) [inline docs/admin/common_tasks.md:56]:
```suggestion
16. Create a POSIX container with the `daos` command.
```

File: docs/admin/common_tasks.md [hunk 16]
```diff
@@ -0,0 +1,110 @@
...
+17. Mount the container with dfuse.
+18. Add a large file that's less than the 500MB pool size into the container.
+19. Call dmg pool query and check that the free size has declined.
+
+## Pool size management
+
+1. Start DAOS server with PMEM + NVMe and format.
+2. Create a pool with the size percentage. For example,
```
Comment (daltonbohning) [inline docs/admin/common_tasks.md:64]:
```suggestion
2. Create a pool with a size percentage. For example,
```

File: docs/admin/common_tasks.md [hunk 17]
```diff
@@ -0,0 +1,110 @@
...
+## Single host setup with PMEM and NVMe
+
+1. Check PMEM and NVMe are discovered by the system. Format and reset them.
+2. Check network configuration. Check the `fabric_iface` you want to use is active.
+3. Install `daos-server` and `daos-client` RPMs.
+4. Generate certificate files.
+5. Copy one of the example configs from `utils/config/examples` to
+`/etc/daos` and adjust it based on the environment. E.g., `access_points`,
+`bdev_class`.
+6. Check that directory where the log files will be created exists. E.g.,
```
[the remark this answers is not in the document]
Comment (shimizukko) [reply docs/admin/common_tasks.md]: Done

File: docs/admin/common_tasks.md [hunk 18]
```diff
@@ -0,0 +1,110 @@
...
+`control_log_file`, `log_file` field in `engines` section.
+7. Start `daos_server`.
+8. Use `dmg config generate` to generate the config file that contains PMEM and
+NVMe.
+9. Define the certificate files in the server config.
+10. Start server with the generated config file.
+11. Check that it's waiting for SCM format. Call dmg storage format.
```
[the remark this answers is not in the document]
Comment (shimizukko) [reply docs/admin/common_tasks.md]: Done

File: docs/admin/common_tasks.md [hunk 19]
```diff
@@ -0,0 +1,110 @@
...
+12. Create a small pool; ~500MB.
```
[the remark this answers is not in the document]
Comment (shimizukko) [reply docs/admin/common_tasks.md]: Done

File: docs/admin/common_tasks.md [hunk 20]
```diff
@@ -0,0 +1,110 @@
...
+13. Define the certificate files in the agent config.
+14. Start agent.
+15. Create a POSIX container with daos command.
```
[the remark this answers is not in the document]
Comment (shimizukko) [reply docs/admin/common_tasks.md]: Done

File: docs/admin/common_tasks.md [hunk 21]
```diff
@@ -0,0 +1,110 @@
...
+18. Call `dmg pool query` and check that the free size has declined.
+
+## Multiple host setup with PMEM and NVMe
+
+1. Check PMEM and NVMe configurations are homogeneous. I.e., same number of
+disks, size, address, etc.
+2. Check network configuration. Check that both the server and the client hosts
+can communicate with the network interface.
+3. Install the same version of `daos-server` and `daos-client` RPMs to all the
+hosts.
```
[the remark this answers is not in the document]
Comment (shimizukko) [reply docs/admin/common_tasks.md]: Right. I updated the wording. Thanks.

File: docs/admin/common_tasks.md [hunk 22]
```diff
@@ -0,0 +1,110 @@
...
+4. Generate certificate files and distribute them to all the hosts.
+5. Copy one of the example configs from `utils/config/examples` to
+`/etc/daos` of one of the server hosts and adjust it based on the environment.
+E.g., `access_points`, `bdev_class`.
+6. Check that directory where the log files will be created exists. E.g.,
```
[the remark this answers is not in the document]
Comment (shimizukko) [reply docs/admin/common_tasks.md]: Done

File: docs/admin/common_tasks.md [hunk 23]
```diff
@@ -0,0 +1,110 @@
...
+`control_log_file`, `log_file` field in `engines` section.
+7. Start `daos_server`.
+8. Use dmg config generate to generate the config file that contains PMEM and
```
[the remark this answers is not in the document]
Comment (shimizukko) [reply docs/admin/common_tasks.md]: Done

File: docs/admin/common_tasks.md [hunk 24]
```diff
@@ -0,0 +1,110 @@
...
+NVMe.
+9. Distribute the config file to `/etc/daos` of all hosts.
+10. Start server on all the hosts.
+11. Check that it's waiting for SCM format. Call `dmg storage format` against all
```
[the remark this answers is not in the document]
Comment (shimizukko) [reply docs/admin/common_tasks.md]: Done

File: docs/admin/common_tasks.md (hunk 14 above)
Comment (shimizukko) [reply docs/admin/common_tasks.md]: Done

File: docs/admin/common_tasks.md (hunk 15 above)
Comment (shimizukko) [reply docs/admin/common_tasks.md]: Done

File: docs/admin/common_tasks.md [hunk 25]
```diff
@@ -0,0 +1,110 @@
...
+server hosts.
+12. Check that the servers are running on all the hosts with `dmg system query
+--verbose`.
+13. Create a small pool; ~500MB.
+14. Define the certificate files in the agent config in the client host.
+15. Start agent.
+16. Create a POSIX container with daos command.
+17. Mount the container with dfuse.
+18. Add a large file that's less than the 500MB pool size into the container.
+19. Call dmg pool query and check that the free size has declined.
```
[the remark this answers is not in the document]
Comment (shimizukko) [reply docs/admin/common_tasks.md]: Done

File: docs/admin/common_tasks.md (hunk 16 above)
Comment (shimizukko) [reply docs/admin/common_tasks.md]: Done

File: docs/admin/common_tasks.md [hunk 26]
```diff
@@ -0,0 +1,110 @@
...
+associated Socket ID.
+- Use `control_log_mask: ERROR` to show only necessary messages in status.
+
+## Multiple server ranks in a single host
+
+- Specify multiple engines under `engines` section. Each section starts with a
+dash `-`.
+- Each engine must use different `fabric_iface_port`. The port numbers should
+differ by at least 100.
+- Each engine must have different `log_file`.
```
[the remark this answers is not in the document]
Comment (shimizukko) [reply docs/admin/common_tasks.md]: Done

**Review by daltonbohning — APPROVED** [review]

**Review by dinghwah — APPROVED** [review]

**Comment by johannlombardi** [comment]
@Michael-Hennecke could you please inspect?

**Pull request merged**